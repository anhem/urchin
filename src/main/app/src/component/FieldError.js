import React, {Component} from 'react'

class FieldError extends Component {

    render() {
        let field = this.props.field;
        let fieldErrors = this.props.fieldErrors || {};
        let errorMessages = fieldErrors[field] ? (fieldErrors[field].map((item, index) =>
            <div className="fieldError" key={index.toString()}>
                {item}
            </div>
        )) : null;

        return (<div>
            {errorMessages &&
            <div className="fieldErrors">
                {errorMessages}
            </div>
            }
        </div>)
    }
}

export default FieldError