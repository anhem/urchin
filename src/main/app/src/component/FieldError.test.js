import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import FieldError from "./FieldError";

describe('FieldError', () => {

    it('match snapshot without error to display', () => {
        expect(toJson(shallow(<FieldError/>))).toMatchSnapshot();
    });

    it('match snapshot with error to display', () => {

        let props = {
            field: 'someField',
            fieldErrors: {
                someField: [
                    'error message',
                    'some other error message'
                ],
                someOtherField: [
                    'error message that should not be displayed'
                ]
            }
        };

        expect(toJson(shallow(<FieldError {...props}/>))).toMatchSnapshot();
    });
});
