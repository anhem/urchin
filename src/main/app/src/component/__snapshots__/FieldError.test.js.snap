// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldError match snapshot with error to display 1`] = `
<div>
  <div
    className="field-errors"
  >
    <div
      className="field-errors__message"
      key="0"
    >
      error message
    </div>
    <div
      className="field-errors__message"
      key="1"
    >
      some other error message
    </div>
  </div>
</div>
`;

exports[`FieldError match snapshot without error to display 1`] = `<div />`;
